frontend/apps/reader/modules/readerdevicestatus.lua

```diff
@@ -12,114 +14,286 @@ local ReaderDeviceStatus = InputContainer:new{
         self.checkLowBattery = function()
-            local threshold = G_reader_settings:readSetting("low_battery_threshold") or 20
             local battery_capacity = powerd:getCapacity()
             if powerd:isCharging() then
-                powerd:setDissmisBatteryStatus(false)
-            elseif powerd:getDissmisBatteryStatus() ~= true and battery_capacity <= threshold then
-                local low_battery_info
-                low_battery_info = ButtonDialogTitle:new {
-                    modal = true,
-                    title = T(_("The battery is getting low.\n%1% remaining."), battery_capacity),
-                    title_align = "center",
-                    title_face = Font:getFace("infofont"),
+                powerd:setDismissBatteryStatus(false)
+            elseif powerd:getDismissBatteryStatus() ~= true and battery_capacity <= self.battery_threshold then
+                powerd:setDismissBatteryStatus(true)
+                UIManager:show(ConfirmBox:new {
+                    text = T(_("Low battery level: %1%\n\nDismiss low battery alarm?"), battery_capacity),
+                    ok_text = _("Dismiss"),
                     dismissable = false,
-                    buttons = {
-                        {
-                            {
-                                text = _("Dismiss"),
-                                callback = function()
-                                    UIManager:close(low_battery_info)
-                                    powerd:setDissmisBatteryStatus(true)
-                                    UIManager:scheduleIn(300, self.checkLowBattery)
-                                end,
-                            },
-                        },
-                    }
-                }
-                UIManager:show(low_battery_info)
-                return
-            elseif powerd:getDissmisBatteryStatus() and battery_capacity > threshold then
-                powerd:setDissmisBatteryStatus(false)
+                    cancel_callback = function()
+                        powerd:setDismissBatteryStatus(false)
+                    end,
+                })
+            elseif powerd:getDismissBatteryStatus() and battery_capacity > self.battery_threshold then
+                powerd:setDismissBatteryStatus(false)
             end
-            UIManager:scheduleIn(300, self.checkLowBattery)
+            UIManager:scheduleIn(self.battery_interval * 60, self.checkLowBattery)
         end
-        self.ui.menu:registerToMainMenu(self)
         self:startBatteryChecker()
-    else
-        self.checkLowBattery = nil
     end
- end
+
+    if not Device:isAndroid() then
+        self.memory_interval = G_reader_settings:readSetting("device_status_memory_interval") or 5
+        self.memory_threshold = G_reader_settings:readSetting("device_status_memory_threshold") or 100
+        self.checkHighMemory = function()
```

Comment:
`checkHighMemoryUsage` (and `checkLowBatteryLevel`) would sound better than `checkHighMemory` (and `checkLowBattery`).